src/cbapi/psc/threathunter/models.py

```diff
@@ -175,3 +177,694 @@ def children(self):
+            # Maybe abstract this into a separate method?
+            resp = cb.get_object(self.urlobject_single.format(model_unique_id))
+            item = resp.get("feedinfo", {})
+            reports = resp.get("reports", [])
+
+        feed_id = item.get("id")
+
+        super(Feed, self).__init__(cb, model_unique_id=feed_id, initial_data=item,
+                                   force_init=False, full_doc=True)
+
+        self._reports = [Report(cb, initial_data=report, feed_id=feed_id) for report in reports]
+
+    def save(self):
+        """Saves this feed on the ThreatHunter server.
+
+        :return: The saved feed
+        :rtype: :py:class:`Feed`
+        """
+        self.validate()
+
+        body = {
+            'feedinfo': self._info,
+            'reports': [report._info for report in self._reports],
+        }
+
+        new_info = self._cb.post_object("/threathunter/feedmgr/v1/feed", body).json()
+        self._info.update(new_info)
+        return self
+
+    def validate(self):
+        """Validates this feed's state.
+
+        :raise InvalidObjectError: if the feed's state is invalid
+        """
+        super(Feed, self).validate()
+
+        if self.access not in ["public", "private"]:
+            raise InvalidObjectError("access should be public or private")
+
+        if not validators.url(self.provider_url):
+            raise InvalidObjectError("provider_url should be a valid URL")
+
+        for report in self._reports:
+            report.validate()
+
+    def delete(self):
+        """Deletes this feed from the ThreatHunter server.
+
+        :raise InvalidObjectError: if `id` is missing
+        """
+        if not self.id:
+            raise InvalidObjectError("missing feed ID")
+
+        self._cb.delete_object("/threathunter/feedmgr/v1/feed/{}".format(self.id))
+
+    def update(self, **kwargs):
+        """Update this feed's metadata with the given arguments.
+
+        >>> feed.update(access="private")
+
+        :param kwargs: The fields to update
+        :type kwargs: dict(str, str)
+        :raise InvalidObjectError: if `id` is missing or :py:meth:`validate` fails
+        :raise ApiError: if an invalid field is specified
+        """
+        if not self.id:
+            raise InvalidObjectError("missing feed ID")
+
+        for key, value in kwargs.items():
+            if key in self._info:
+                self._info[key] = value
+
+        self.validate()
+
+        url = "/threathunter/feedmgr/v1/feed/{}/feedinfo".format(self.id)
+        new_info = self._cb.put_object(url, self._info).json()
+        self._info.update(new_info)
+
+        return self
+
+    @property
+    def reports(self):
+        """Returns a list of :py:class:`Report` associated with this feed.
+
+        :return: a list of reports
+        :rtype: list(:py:class:`Report`)
+        """
+        return self._cb.select(Report).where(feed_id=self.id)
+
+    def replace(self, reports, append=False):
```

Comment:
I would like to split this into two functions - one to replace all reports and one to add (append) reports. I feel as if this is too easy to shoot yourself in the foot by forgetting to add the `append` argument

Proposed naming would be `replace_reports` and `append_reports` (just to make it clear what you're appending and replacing)

Reply:
Sounds good! I'll make that split.